### ISSUE TITLE
[no-unsed-vars] : Add annotation to ignore exports

### ISSUE THREAD

**Issue by awilkins** [eslint/eslint#10312]
**What rule do you want to change?**

`no-unsed-vars`

**Does this change cause the rule to produce more or fewer warnings?**

More.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New comment annotation.

**Please provide some example code that this change will affect:**

```js

function foo() {
  // do stuff but don't call bar()
}

// bar is meant to be private but we exported it to unit test it
function bar() {
 // do more stuff
}

/* eslint ignore-export bar */
module.exports = foo, {bar}

```

**What does the rule currently do for this code?**

No unused vars detected.

**What will the rule do after it's changed?**

Detect bar as unused.

---

Rationale : 

There are lots of awkward ways of [testing private methods](https://stackoverflow.com/questions/22097603/unit-testing-of-private-functions-with-mocha-and-node-js) in JS code, but exporting functions is the easiest way to get access to them for tests. This however obscures instances where they are not used in-module by the code that should be calling them.
